lib/postgrex/protocol.ex

```diff
@@ -81,12 +81,29 @@ defmodule Postgrex.Protocol do
     timeout = opts[:timeout] || @timeout
     ping_timeout = Keyword.get(opts, :ping_timeout, timeout)
     sock_opts = [send_timeout: timeout] ++ (opts[:socket_options] || [])
-    ssl? = opts[:ssl] || false
     types_mod = Keyword.fetch!(opts, :types)
     disconnect_on_error_codes = opts[:disconnect_on_error_codes] || []
     target_server_type = opts[:target_server_type] || :any
     disable_composite_types = opts[:disable_composite_types] || false
 
+    {ssl_opts, opts} =
+      case Keyword.pop(opts, :ssl, false) do
+        {false, opts} ->
+          {nil, opts}
+
+        {true, opts} ->
+          Logger.warning(
+            "setting ssl: true on your database connection offers only limited protection, " <>
+              "as the server's certificate is not verified. Set \"ssl: [cacertfile: path/to/file]\" instead"
+          )
+
+          # Read ssl_opts for backwards compatibility
+          Keyword.pop(opts, :ssl_opts, [])
```

Comment:
👋 

Sorry for posting on an old PR, but should the default be `verify_peer: :none`? it seems like, by default, `:ssl` still tries to verify the peer.

```elixir
iex(1)> :ssl.connect(~c"google.com", 443, [])
{:error, {:options, :incompatible, [verify: :verify_peer, cacerts: :undefined]}}
```

Reply:
I think previous OTP defaulted to :verify_none and more recent to :verify_peer and so I believe the idea behind this legacy `ssl: true` was to keep whatever that particular OTP defaulted to however `ssl: list` would force secure defaults.

Reply:
Do you know if this change broke any apps? Cause if the warning implies less security we should change the warning not intentionally make things less secure. Sorry I don't have an easy way to check how it all works right now.

Reply:
No, I don't think it broke any apps. My suggestion was purely a QoL improvement :)